### PR TITLE
Adds pretrained model fetching from static.menpo.org

### PR DESCRIPTION
**Pull request by jabooth**
This PR adds `menpofit.io`, a module that contains code for fetching pretrained fitters from `static.menpo.org`.

Fitters that are downloaded are stored in `data`, and so the download only needs to be done the first time a particular pretrained model is requested.

Note that the actual contents of the pickle that is download is expected to be an invokable callable that will return a fitter that is ready to use.

Models are exposed under the relevant menpofit namespace. Currently only one model is exposed, and should be used as so:

```py
from menpofit.aam import load_balanced_frontal_face_fitter
fitter = load_balanced_frontal_face_fitter()
fitter.fit_from_bb(...)
```
### TODO

- [x] Check everything works for Python 2.7
- [x] Update the model to a better one (@nontas)
- [x] Update the docstring on `load_balanced_frontal_face_fitter()` to explain some details about the model offered

Note that this is a self contained addition to menpofit, and can be released without changing any other dependencies. A good proportion of `menpofit.io` will probably be implemented in `menpo` (see https://github.com/menpo/menpo/pull/710). In the interest of getting something working pre-CVPR, we will go with this simpler approach for now.

### CODE REVIEW

File: menpofit/aam/pretrained.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+from menpofit.io import load_fitter
+
+
+def load_balanced_frontal_face_fitter():
```
Comment (nontas) [inline menpofit/aam/pretrained.py]: Done

**Comment by nontas** [comment]
After running (about 1300 experiments!!), we decided on the balanced model. For future reference, the parameters are

```
diagonal=110
scales=(0.5, 1.0)
patch_shape=[(13, 13), (13, 13)]
holistic_features=fast_dsift
n_shape=[5, 20]
n_appearance=[30, 150]
lk_algorithm_cls=WibergInverseCompositional
```

and `sampling` is:

```
import numpy as np

patch_shape = (13, 13)
sampling_step = 4

sampling_grid = np.zeros(patch_shape, dtype=np.bool)
sampling_grid[::sampling_step, ::sampling_step] = True
sampling = [sampling_grid, sampling_grid]
```

The model size is about 66MB and loading time about 100ms. I found that by fitting using `max_iters=[25, 5]` or even `max_iters=[20, 5]` the fitting accuracy is good enough and the fitting time is about 258ms and 237ms respectively. 

The only remaining question is how to encode the default `max_iters` value. I'm not sure that we should change it on the `MultiScaleNonParametricFitter` since it is a general class. But I don't know what otehr options we have...

**Comment by jabooth** [comment]
Amazing work @nontas!

> The only remaining question is how to encode the default max_iters value.

Not sure I follow? For this model we just set this kwarg in the partial over the fitter to encode that no?
https://gist.github.com/jabooth/85ee24abbdd7103aa3281487bb99a3e8

So this would be something like
```py
create_fitter = partial(LucasKanadeAAMFitter, aam, 
                        lk_algorithm_cls=WibergInverseCompositional, 
                        sampling=sampling, max_iters=[25, 5])
```

**Comment by nontas** [comment]
No  `max_iters` is an argument on `fitter.fit_from_bb()` and `fitter.fit_from_shape()`

**Comment by jabooth** [comment]
Maybe we need a structure in menpofit that allows us to capture parameters on both fitter construction and fitter fitting then:
```python
class PickleFitterWrapper(object):
    
    def __init__(self, fitter_cls, fitter_args, fitter_kwargs, 
                 fit_from_bb_kwargs, fit_from_shape_kwargs):
        self.wrapped_fitter = fitter_cls(*fitter_args, **fitter_kwargs)
        self.fit_from_bb_kwargs = fit_from_bb_kwargs
        self.fit_from_shape_kwargs = fit_from_shape_kwargs
    
    def fit_from_bbox(self, image, bounding_box, **kwargs):
        final_kwargs = self.fit_from_bb_kwargs.copy() 
        final_kwargs.update(kwargs)
        return self.wrapped_fitter.fit_from_bb(image, bounding_box, 
                                               **final_kwargs)
    
    def fit_from_shape(self, image, initial_shape, **kwargs):
        final_kwargs = self.fit_from_shape_kwargs.copy() 
        final_kwargs.update(kwargs)
        return self.wrapped_fitter.fit_from_shape(image, initial_shape,
                                                  **final_kwargs)

```
usage:
```py
from menpofit.io import PickleFitterWrapper

# kwargs for fitter construction
fitter_kwargs = dict(lk_algorithm_cls=WibergInverseCompositional, 
                     sampling=sampling)

# kwargs for fitter.fit_from_{bb, shape}
# (note here we reuse the same kwargs twice)
fit_kwargs = dict(max_iters=[25, 5])

fitter_wrapper = partial(PickleFitterWrapper, LucasKanadeAAMFitter, aam, 
                         fitter_kwargs, fit_kwargs, fit_kwargs)

menpo.io.export_pickle(fitter_wrapper, 'fitter.pkl')
```
and at load time:
```py
fitter = menpo.io.import_pickle('fitter.pkl')()
fitter.fit_from_bb(...) # has saved kwargs (but can still be overwritten)
fitter.wrapped_fitter  # access to the original fitter if needed.
```
 @nontas @patricksnape ?

**Comment by nontas** [comment]
That looks fine to me! Once this is done, I will train the final AAM model.

**Comment by patricksnape** [comment]
Sure,  I mean this is just for the shipped models so as long as documentation is there it's fine

**Comment by jabooth** [comment]
Ignore jenkins, he's drunk. This is ready to come in.

**Comment by patricksnape** [comment]
OK, looks good to me - assume you tested it with a real model?

**Comment by jabooth** [comment]
@patricksnape yeah locally tested with a model @nontas gave me a few days ago. Only unknown is if that model works OK on Windows (current plan is to ship two models one for py2 one for py3, think we resolved all win/unix issues with this):

https://github.com/menpo/menpo/blob/v0.7.1/menpo/io/output/pickle.py#L31

Will check now

**Comment by jabooth** [comment]
Confirmed that our test model works between windows/mac - merging.

File: menpofit/io.py [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+    Returns
+    -------
+    fitter : `Fitter`
+        A pre-trained menpofit `Fitter` that is ready to use.
+    """
+    path = path_of_fitter(name)
+    if not path.exists():
+        print('Downloading {} fitter'.format(name))
+        url = url_of_fitter(name)
+        print(url, path)
```
Comment (patricksnape) [inline menpofit/io.py]: @jabooth Is this print supposed to be here?

Comment (jabooth) [inline menpofit/io.py]: @patricksnape it's definitely more of a debug statement that I forgot that I forgot to remove, apologies.
Not the end of the world being there, feel free to spice it up or remove it if you want to!

**Pull request merged**